src/components/AddFavoriteMovieList/types.ts

```diff
@@ -0,0 +1,8 @@
+import { IMovieData } from "@api/types";
+import { Dispatch, SetStateAction } from "react";
+
+export interface IAddFavoriteMoviesList {
+  moviesDate: Array<IMovieData> | undefined;
```

Comment:
опциональность параметра можно указать с помощью `?`
```suggestion
  moviesDate?: Array<IMovieData>;
```